### DISCUSSION TITLE
How do I update a component property and access it in another component method?

### DISCUSSION THREAD

**Discussion by tangaye** [alpinejs/alpine#1074] [1. Help]
Hello everyone. I simply want to update the property of an alpine component from another component method and be able to  access the updated component property.
```javascript
function test (){
    return {
        updateMe: [],
        update() {
            this.updateMe = ['1', 2, true]
        }
    }
}
```
The problem is when I call `window.test().update()` or `test().update()`, the property doesn't stay updated. It reverts to what it was previously assigned.

**Comment by SimoTod** [comment]
That piece of code doesn't do what you need. Every time you call test(), you'll get a new copy of the object. Also, you need to go through the Alpine proxies in order to make everything reactive.
